src/main/kotlin/lotto/domain/WinningStatisticsCalculator.kt

```diff
@@ -0,0 +1,12 @@
+package lotto.domain
+
+import lotto.controller.WinningNumbers
+
+class WinningStatisticsCalculator {
+    fun calculate(winningNumbers: WinningNumbers, tickets: Tickets): Pair<WinningStatistics, Double> {
+        val statistics = WinningStatistics()
+        tickets.checkMatches(winningNumbers, statistics)
```

Comment:
What if `tickets.checkMatches()` returned a `WinningStatistics`?
What advantages do you see in moving the responsibility there?

Reply:
I considered it, but I thought it would be weird to return WinningStatistics from Tickets.
But if I refactor this code as you mentioned, it would be cleaner than the way I created a separate statistics object inside WinningStatisticsCalculator.